src/main/java/org/folio/fqm/service/EntityTypeFlatteningService.java

```diff
@@ -48,54 +51,75 @@ public EntityType getFlattenedEntityType(UUID entityTypeId, boolean doFinalRenam
       .name(originalEntityType.getName())
       ._private(originalEntityType.getPrivate())
       .defaultSort(originalEntityType.getDefaultSort())
-      .columns(originalEntityType.getColumns()) // TODO: probably remove
       .idView(originalEntityType.getIdView())
       .customFieldEntityTypeId(originalEntityType.getCustomFieldEntityTypeId())
       .labelAlias(originalEntityType.getLabelAlias())
       .root(originalEntityType.getRoot())
       .groupByFields(originalEntityType.getGroupByFields())
-      .sourceView(originalEntityType.getSourceView()) // Possibly unneeded
-      .sourceViewExtractor(originalEntityType.getSourceViewExtractor()); // Possibly unneeded
+      .sourceView(originalEntityType.getSourceView())
+      .sourceViewExtractor(originalEntityType.getSourceViewExtractor());
+
+    Map<String, String> renamedAliases = new HashMap<>(); // <oldName, newName>
+    String aliasPrefix = sourceFromParent == null ? "" : sourceFromParent.getAlias() + ".";
+    for (EntityTypeSource source : originalEntityType.getSources()) {
+      // Update alias
+      String oldAlias = source.getAlias();
+      String newAlias = aliasPrefix + oldAlias;
+      renamedAliases.put(oldAlias, newAlias);
+    }
 
-    List<EntityTypeColumn> finalColumns = new ArrayList<>();
     Set<String> finalPermissions = new HashSet<>(originalEntityType.getRequiredPermissions());
+    Stream.Builder<Stream<EntityTypeColumn>> columns = Stream.<Stream<EntityTypeColumn>>builder()
+      .add(copyColumns(sourceFromParent, originalEntityType, renamedAliases));
```

Comment:
It occurred to me that this method ends up spending quite a bit of time just creating lists, only to copy the list contents into another list. Rinse and repeat. I refactored the column handling, so that it keeps them in streams and generally doesn't bother actually dumping them into the final list until the very end, to make this a little better. It'll still probably thrash the garbage collector a bit and allocate a lot more stuff than necessary, but it should be at least a tiny bit better